eclipse-collections/src/main/java/org/eclipse/collections/impl/lazy/FlatCollectIterable.java

```diff
@@ -78,6 +79,36 @@ public V detect(Predicate<? super V> predicate)
     }
 
     @Override
+    public Optional<V> detectOptional(Predicate<? super V> predicate)
+    {
+        V[] result = (V[]) new Object[1];
+        Iterate.anySatisfy(this.adapted, each -> {
+            if (each == null)
+            {
+                throw new NullPointerException();
+            }
+            Iterable<V> iterable = this.function.valueOf(each);
+            if (iterable == null)
+            {
+                throw new NullPointerException();
+            }
+            return Iterate.anySatisfy(iterable, each1 -> {
+                if (predicate.accept(each1))
+                {
+                    if (each1 == null)
+                    {
+                        throw new NullPointerException();
+                    }
+                    result[0] = each1;
```

Comment:
We have to differentiate the case we detect null (we only know it by checking `each1` == null after evaluating predicate.accept(each1)) vs the case we didn't find anything (result[0] stays null).